### PR TITLE
Heroku and Express setup

### LINKED ISSUE #40: Deploy hammer-cli generated project to heroku
As a user, I would like my auto generated hammer-cli project to be deployed to heroku. 

- [ ] Verify the global prereq (created in #33) questions now includes a question about heroku account.
- [ ] Verify hammer-cli prompts the user for their username and password
- [ ] Verify hammer-cli can connect to the heroku API
- [ ] Verify that the hammer-cli project is deployed to heroku

### PR DESCRIPTION
**Pull request by nischayv**
closes #40 , #42 , #45 
This PR makes the following changes- 

- Cleans up github client and creates a git util
- Removes all the template strings spread across the project in different forms and puts them in one unified location
- Add functionality for ExpressJS
- Allows adding of env vars to travis builds
- Add functionality to deploy to Heroku
- Generates the .travis.yml files dynamically based on user choices
- Adds additional linting rules
- Lots of tests!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "fs-extra": "4.0.2",
     "inquirer": "1.2.3",
     "is-valid-path": "0.1.1",
+    "js-yaml": "^3.10.0",
```
[the remark this answers is not in the document]
Comment (nischayv) [reply package.json]: Thanks :)

File: src/utils/file.js [hunk 2]
```diff
@@ -16,15 +23,30 @@ export function createPackageJson(config) {
...
 }
 
 /**
  * Generate a simple index.js file
  */
-export function createIndexFile(folderName) {
-  fs.writeFileSync(`${folderName}/src/index.js`, constants.indexJS.fileContents);
+export function createIndexFile(folderName, usingExpress) {
+  if (usingExpress) {
+    fs.writeFileSync(
```
Comment (enclark) [inline src/utils/file.js]: We should put this in a new file with other express-related logic, so when we do have to change the code to fit a new version, it is all in one place.

File: package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (jackcmeyer) [reply package.json]: Why is there not a dependency for express? Is this intentional?

[the remark this answers is not in the document]
Comment (nischayv) [reply package.json]: @jackcmeyer Yes. Our project doesn't need Express. This is the package.json for the cli

**Pull request merged**